pkg/cli/admin/inspectalerts/inspectalerts.go

```diff
@@ -121,24 +124,24 @@ func getWithBearer(ctx context.Context, getRoute RouteGetter, namespace, name st
 		uris = append(uris, uri.String())
 		req, err := http.NewRequest("GET", uri.String(), nil)
 		if err != nil {
-			return nil, err
+			return nil, 0, err
 		}
 
 		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", bearerToken))
 
 		resp, err := client.Do(req)
 		if err != nil {
-			return nil, err
+			return nil, 0, err
 		}
 		defer resp.Body.Close()
 
 		body, err := io.ReadAll(resp.Body)
 		if err != nil {
-			return nil, err
+			return nil, 0, err
 		}
 
-		return body, err
+		return body, resp.StatusCode, err
```

Comment:
Can we put the `!= http.StatusOK` check here, and return an `error` in the case where we failed?  Having some details on why would also be interesting, but dumping HTML to the terminal might be too chatty.  I wonder if we could set `Accept` to say "we want JSON back, but if you can't do that, `text/plain`" to get something more terminal-compatible?  Or maybe Thanos isn't that media-type aware?  Or maybe it's the router that's giving us an error (like "none of the Pods backing this Service are `Ready=True`", and not Thanos?)?

Reply:
I did it this way because `getWithBearer` sounded like a low-level HTTP method which I would not expect to have opinions on status codes (maybe something would want to call it and handle retries or something).

Router's "service is currently down" page is what I assume was happening.

Reply:
I'm not entirely convinced about high-level vs. low-level, but the 8fd03afed69225b logging gives folks calling with `-v8` or higher access to the body, and that's the main think I was hoping for.  And `getWithBearer` is internal, so we can shift things around later if we change our minds.  Feel free to mark this thread resolved :)

Reply:
I would second putting the check here. No reason really to extend the interface, but its really a nit, feel free to ignore.

As for thanos responses, it implements the [Prometheus query api](https://prometheus.io/docs/prometheus/latest/querying/api/).  There it says `The API response format is JSON`

Reply:
I have put the check to `getWithBeared` as you advise, but while I did it, I noticed that the iteration over ingress hosts from a route does not make much sense the way it was done - the iteration aborted on failure and continued on success, so it actually required all options to succeed and then returned the last one. I guess this would not be an issue normally because there would be just a single option, but still changed the method to search for a first success, and only return with an error if all options failed with an error.

Reply:
Yeah good catch :+1: